app/src/main/java/org/mozilla/fenix/compose/button/TextButton.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.button
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Base component for text-only buttons.
+ *
+ * @param text The button text to be displayed.
+ * @param onClick Invoked when the user clicks on the button.
+ * @param modifier [Modifier] Used to shape and position the underlying [androidx.compose.material.TextButton].
+ * @param textColor [Color] to apply to the button text.
+ */
+@Composable
+fun TextButton(
+    text: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    textColor: Color = FirefoxTheme.colors.textAccent,
+) {
+    androidx.compose.material.TextButton(
+        onClick = onClick,
+        modifier = modifier,
+    ) {
+        Text(
+            text = text.uppercase(),
+            style = TextStyle(
+                color = textColor,
+                fontSize = 14.sp,
+                fontWeight = FontWeight.Medium,
+                letterSpacing = 0.25.sp,
+                lineHeight = 14.sp,
+            ),
```

Comment:
One day we probably want to group these `TextStyle` into something like Header, Title, Description, etc if we can find some kind of commonality within the component system, but ultimately needs to be defined within the design system.

Reply:
Yes!! My current impression is that we'll have some good progress on the components and then we'll start on a `TextStyle` library and eventually a `Shapes` library.

Reply:
Feel free to file an issue for the `TextStyle` and work on it. It appears ready, but you should double check. https://www.figma.com/file/pEyGeE4KV5ytYHeXMfLcEr/Mobile-Styles?node-id=861%3A54432

Similar to how we added usage comments for the design color tokens. I would also make a note to also include the usage notes that are included in the Figma usage section. I would just copy it verbatim since I think all the documentations related to these component/color usages should be defined in Figma and managed properly from there as the source.

Reply:
https://github.com/mozilla-mobile/fenix/issues/25379